packages/core/src/components/widgets/CheckboxWidget.tsx

```diff
@@ -58,12 +63,18 @@ function CheckboxWidget<T = any, S extends StrictRJSFSchema = RJSFSchema, F exte
   );
   const description = options.description ?? schema.description;
 
+  const richDescription = uiOptions.enableMarkdownInDescription ? (
+    <Markdown options={{ disableParsingRawHTML: true }}>{description || ''}</Markdown>
+  ) : (
+    description || ''
+  );
+
```

Comment:
@dejanbelusic after looking around it turns out that every theme's `CheckboxWidget` has a similar description rendering logic in it. We suggest creating a new component in the `@rjsf/utils` package that wraps this logic as follows and then updating every `FieldDescriptionTemplate` to use it:

```tsx
export interface RichDescriptionProps<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any> {
  /** The description of the field being rendered */
  description: ReactNode;
  /** The uiSchema that was passed to field */
  uiSchema?: UiSchema<T, S, F>;
  /** The `registry` object */
  registry: Registry<T, S, F>;
}
export default function RichDescription<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({ description, registry, uiSchema = {}}: RichDescriptionProps<T,S,F>) {
  const { globalUiOptions } = registry;
  const uiOptions = getUiOptions<T, S, F>(uiSchema, globalUiOptions);
  if (uiOptions.enableMarkdownInDescription) {
    return <Markdown options={{ disableParsingRawHTML: true }}>{description || ''}</Markdown>
 }
 return description;
}
```

With all the `FieldDescriptionTemplates` calling it similarly to the change that would be made in `@rjsf/core`'s implementation:
```
export default function DescriptionField<
  T = any,
  S extends StrictRJSFSchema = RJSFSchema,
  F extends FormContextType = any
>(props: DescriptionFieldProps<T, S, F>) {
  const { id, description, uiSchema, registry } = props;
  if (!description) {
    return null;
  }
  return (
   <div id={id} className='field-description'>
      <RichDescription description={description} uiSchema={uiSchema} registry={registry} />
    </div>
  );
}
```

Reply:
Hi @heath-freenome, I see your point and I agree. I can create a new component `RichDescription` as suggested, place it in the `@rjsf/utils`, and then use it in every `DescriptionField` component. 

When looking into `DescriptionField` components, I can see that one in `@rjsf/antd` packages is wrapped in a `span`, the one in `@rjsf/bootstrap-4` is wrapped in two `div`s, the one in  `@rjsf/chakra-ui` package is wrapped in `Text` component... What I'm saying is every theme has similar, but different implementation. Do I leave them as is?